ddspipe_core/src/cpp/types/dynamic_types/schema_msg.cpp

```diff
@@ -60,53 +61,59 @@ struct TreeNodeType
 
 // Forward declaration
 std::string type_kind_to_str(
-        const fastrtps::types::DynamicType_ptr& type);
+        const fastdds::dds::DynamicType::_ref_type& type);
 
-fastrtps::types::DynamicType_ptr container_internal_type(
-        const fastrtps::types::DynamicType_ptr& dyn_type)
+fastdds::dds::DynamicType::_ref_type container_internal_type(
+        const fastdds::dds::DynamicType::_ref_type& dyn_type)
 {
-    return dyn_type->get_descriptor()->get_element_type();
+    fastdds::dds::TypeDescriptor::_ref_type type_descriptor {fastdds::dds::traits<fastdds::dds::TypeDescriptor>::make_shared()};
+    const auto ret = dyn_type->get_descriptor(type_descriptor);
```

Comment:
In other cases I would agree with you but this is returning a ReturnCode and I don't see the point of explicitly saying it, I think using auto still makes it clear enough. Additionally, using auto in this context doesn't compromise readability and helps avoid repeating the type explicitly, keeping the code more concise. Anyway, I've tried to use the format `if (fastdds::dds::RETCODE_OK != ...` directly instead of previously defining a ReturnCode object.